app/src/main/java/io/homeassistant/companion/android/widgets/ButtonWidget.kt

```diff
@@ -0,0 +1,135 @@
+package io.homeassistant.companion.android.widgets
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import android.widget.RemoteViews
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.common.dagger.GraphComponentAccessor
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+class ButtonWidget : AppWidgetProvider() {
+    companion object {
+        private const val TAG = "ButtonWidget"
+        private const val CALL_SERVICE = "CALL_SERVICE"
+    }
+
+    private lateinit var domain: String
+    private lateinit var service: String
+    private lateinit var serviceData: String
```

Comment:
These shouldn't be a the class level, they should be variables in the methods they are used.

Reply:
Done, these are only in the specific method.